content/docs/gke/private-clusters.md

```diff
@@ -7,60 +7,176 @@ weight = 5
+     address becomes an administrator in the configuration of your Kubeflow 
+     deployment.
+
+1. Enable private clusters by editing `${KFAPP}/gcp_configs/cluster-kubeflow.yaml` and updating the following two parameters:
 
     ```
     privatecluster: true
     gkeApiVersion: v1beta1
     ```
+1. Remove components which are not useful in private clusters:
+
+    ```
+    cd ${KFAPP}/ks_app
+    ks component rm cert-manager
+    ```
 1. Create the deployment:
 
     ```
     cd ${KFAPP}
     kfctl apply platform
     ```
 
-1. To set up ingress to the cluster, it is recommended to use your custom domain instead of Cloud Endpoints. cert-manager cannot be used to create HTTPS certificates because cert-manager needs to talk to LetsEncrypt to get the certificate and that is not possible in a private cluster setting. Obtain the HTTPS certificates for your ${FQDN} and create a k8s secret with it. Assuming your cert and key are present in files named tls.crt and tls.key, create a secret using the following command:
+   * If you get an error **legacy networks not supported** Follow the 
+     [troubleshooting guide]( /docs/gke/troubleshooting-gke/#legacy-networks-are-not-supported) to create a new network.
 
-    ```
-    kubectl create secret generic --namespace=${NAMESPACE} envoy-ingress-tls --from-file=tls.crt=tls.crt --from-file=tls.key=tls.key
-    ```
+     * You will need to manually create the network as a work around for [kubeflow/kubeflow#3071](https://github.com/kubeflow/kubeflow/issues/3071)
+
+       ```
+       cd ${KFAPP}/gcp_configs
+       gcloud --project=${PROJECT} deployment-manager deployments create ${KFAPP}-network --config=network.yaml
+       ```
+
+     * Then edit **gcp_config/cluster.jinja** to add a field **network** in your cluster
+     
+       ```
+       cluster:
+          name: {{ CLUSTER_NAME }}
+          network: <name of the new network>
+       ```
+   
+     * To get the name of the new network run
+      
+       ```
+       gcloud --project=cloud-ml-dev compute networks list
+       ``` 
+
+       * The name will contain the value ${KFAPP}
 
 1. Update iap-ingress component parameters:
 
     ```
-    cd ${KFAPP}/ks_app
-    ks param set iap-ingress hostname ${FQDN}
+    cd ${KFAPP}/ks_app    
     ks param set iap-ingress privateGKECluster true
     ```
 
-1. Create an A record in your DNS management service to point ${FQDN} to the static IP address which was created by deployment manager. It can be found in `gcloud compute addresses list`.
+1. Obtain an HTTPS certificates for your ${FQDN} and create a k8s secret with it. 
 
-1. Update the various ksonnet components to use `gcr.io` images instead of dockerhub images:
+   * You can create a self signed cert using [kube-rsa](https://github.com/kelseyhightower/kube-rsa)
 
-    ```
-    cd ${KFAPP}/ks_app
-    ${KUBEFLOW_SRC}/scripts/gke/use_gcr_for_all_images.sh
-    ```
+      ```
```

Comment:
Same indentation issue with this code block - needs an extra level of indentation in the source.